src/lib/services/validator/lcd.ts

```diff
@@ -34,3 +47,67 @@ export const getValidatorDataLcd = (
       `${endpoint}/cosmos/staking/v1beta1/validators/${encodeURI(validatorAddr)}`
     )
     .then(({ data }) => parseWithError(zValidatorResponseLcd, data).validator);
+
+export const getValidatorStakingProvisionsLcd = async (
+  endpoint: string,
+  chain: string
+) => {
+  if (chain === "osmosis") {
+    const [mintParams, epochProvisions] = await Promise.all([
+      getMintParamsLcd(endpoint, chain),
+      getEpochProvisionsLcd(endpoint, chain),
+    ]);
+
+    const stakingProportion =
+      mintParams.params.distributionProportions?.staking;
+
+    if (!stakingProportion) throw new Error("Staking proportion not found");
+
+    const epochStakingProvisions =
+      epochProvisions.epochProvisions.mul(stakingProportion);
+
+    const epochIndentifier = mintParams.params.epochIdentifier;
+
+    if (epochIndentifier === "hour")
+      return { stakingProvisions: epochStakingProvisions.mul(8760) };
+
+    if (epochIndentifier === "day")
+      return { stakingProvisions: epochStakingProvisions.mul(365) };
+
+    if (epochIndentifier === "week")
+      return { stakingProvisions: epochStakingProvisions.mul(52) };
+
```

Comment:
What if `epochIndentifier` is not all of above? Should we throw an error.

Reply:
Maybe yes, we can throw the error as well.